### DISCUSSION TITLE
Reading adducts like [M+H]+

### DISCUSSION THREAD

**Discussion by asmirn1** [rdkit/rdkit#3843] [Q&A]
Hi,

I wonder if RDKit can read the adduct information like "[M+H]+" and automatically determine the number of molecules, mass difference, and charge?

If not, are there any other packages that can do that?

Thank you

**Comment by greglandrum** [comment]
@asmirn1 : I don't recognize that format. What do you mean by "adduct information"?

**Comment by asmirn1** [comment]
@greglandrum 

In the liquid chromatography coupled to the mass spectrometry, there are certain reactions when a charged adduct of a molecule is formed.

A few examples of such adducts: [M+H]+, [M+H2O]+, [2M+H]+, [M-H]-, where M is the original molecule. Each adduct is characterized by 
- the mass difference (i.e. the mass of H, H2O, -H in the examples above), 
- number of molecules (i.e. the number in front of M)
- charge (e.g. +1 or -1)

I'd like to parse an adduct name (e.g. "[M+H]+") and retrieve the mass difference, the number of molecules, and the charge.

I was wondering if RDKit already has such a functionality.

**Reply by greglandrum** [reply]
Hi @asmirn1,
Looks like I forgot to reply to this... sorry.
No, we don't have anything like that as part of the RDKit. It looks rather specific to mass spec analysis, so doesn't seem like a good match for the core RDKit itself.

**Reply by asmirn1** [reply]
@greglandrum Thank you for looking into it anyway.
